src/server/game/Spells/SpellInfo.cpp

```diff
@@ -3901,7 +3906,9 @@ std::vector<SpellPowerCost> SpellInfo::CalcPowerCost(Unit const* caster, SpellSc
             powerCost += CalculatePct(powerCost, (*i)->GetAmount());
         }
 
-        if (power->PowerType == POWER_HEALTH)
+        if (power->PowerType == POWER_MANA)
+            powerCost = std::max(0.0f, (float)powerCost * (1.0f + caster->GetManaCostMultiplier())); // Ensure >= 0
```

Comment:
There is no requirement that cost must be >= 0, costs can actually be negative (and grant power)

This probably also makes the change above (comparison against -1) incorrect

Reply:
So if a resto druid uses heart of the wild together with innervate, and then cast a spell (e.g. regrowth with cost 240 mana), he will get 120 extra mana ?

Reply:
Resto druid does not have access to Restoration Affinity talent so they cannot benefit from cost reduction with Heart of the Wild

Reply:
Balance druid*

Reply:
Nonhealers cannot be given innervate

Reply:
Well I don't know the game well enough to give other examples.
Since the idea here is to "Reduce mana", it doesn't sound logical to give the player mana from it.

Do you have an example where multiple spells can be applied and the combined mana reduction is above 100%?